### PR TITLE
Feature instance qr code

### PR DESCRIPTION
**Pull request by singhsanket143**
The pull request is implementation for the feature listed in issue #16623. Implementation of the backend of the QR code functionality of the application has been done in order to get a QR Code of the instance URL.

Overall show page of instance
image

The button to trigger the qr code modal
image

QR Code modal
image

### CODE REVIEW

File: client/views/helpers/nav/nav.js [hunk 1]
```diff
@@ -35,7 +35,12 @@ Template.nav.events({
       Blaze.remove(popoverTemplate);
     });
   },
+  'click #navCode': function(event, template) {
+    const parentNode = document.getElementById('nav');
+    popoverTemplate = Blaze.renderWithData(Template.qr_code, {
+      link: `https://questions.dev.berkmancenter.org/list/${template.data.slug}`
```
Comment (peter-hank) [inline client/views/helpers/nav/nav.js]: @singhsanket143 can you replace the hardcoded domain name with some preconfigured environment variable?

File: client/views/helpers/qr_code/qr_code.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<template name="qr_code">
+	<div id="darker"></div>
+	<div class="qr_codeouter">
+    {{> close_button}}
+
+		<div class="modaltopmessage">
+			<h2>Scan the QR code to open the corresponding Instance in your device</h2>
```
Comment (peter-hank) [inline client/views/helpers/qr_code/qr_code.html]: @singhsanket143 `Instance` > `instance`

File: client/views/helpers/qr_code/qr_code.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<template name="qr_code">
```
Comment (peter-hank) [inline client/views/helpers/qr_code/qr_code.html:1]: @singhsanket143 can you fix the style of this template?

File: client/views/helpers/qr_code/qr_code.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.modaltopmessage > h2 {
+  line-height: 25px;
+  margin-top: 5px;
+  color: black;
+}
+
+.qr_codeouter {
```
Comment (peter-hank) [inline client/views/helpers/qr_code/qr_code.scss:7]: @singhsanket143 can you fix the code style of this file?

**Review by peter-hank — CHANGES_REQUESTED** [review]

File: .meteor/packages [hunk 5]
```diff
@@ -45,3 +45,5 @@ dispatch:mocha-phantomjs
 xolvio:cleaner
 reactive-var
 juliancwirko:s-alert
+panter:qrcode
+juliancwirko:s-alert
```
Comment (peter-hank) [inline .meteor/packages]: @singhsanket143 it's duplicated

File: client/views/helpers/qr_code/qr_code.html (hunk 3 above)
Comment (singhsanket143) [reply client/views/helpers/qr_code/qr_code.html:1]: @peter-hank Updated the file

File: client/views/helpers/nav/nav.js (hunk 1 above)
Comment (singhsanket143) [reply client/views/helpers/nav/nav.js]: @peter-hank Updated the domain name

File: client/views/helpers/qr_code/qr_code.scss (hunk 4 above)
Comment (singhsanket143) [reply client/views/helpers/qr_code/qr_code.scss:7]: @peter-hank Updated the file

File: client/views/helpers/qr_code/qr_code.html (hunk 2 above)
Comment (singhsanket143) [reply client/views/helpers/qr_code/qr_code.html]: @peter-hank Updated

File: .meteor/packages (hunk 5 above)
Comment (singhsanket143) [reply .meteor/packages]: @peter-hank I have now rectified this one also

File: .vscode/typings/meteor.d.ts [hunk 6]
```diff
@@ -0,0 +1,2013 @@
+// Generated by typings
```
Comment (peter-hank) [inline .vscode/typings/meteor.d.ts]: @singhsanket143 can you remove this file from the repository?

**Comment by peter-hank** [comment]
@singhsanket143 `public/.DS_Store` can you remove this file from the repository?

File: client/views/helpers/nav/nav.js [hunk 7]
```diff
@@ -35,7 +35,12 @@ Template.nav.events({
       Blaze.remove(popoverTemplate);
     });
   },
+  'click #navCode': function(event, template) {
+    const parentNode = document.getElementById('nav');
+    popoverTemplate = Blaze.renderWithData(Template.qr_code, {
+      link: Meteor.absoluteUrl()+`/list/${template.data.slug}`
```
Comment (peter-hank) [inline client/views/helpers/nav/nav.js]: @singhsanket143 can you add a space before and after `+`?

File: client/views/helpers/qr_code/qr_code.scss [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  display: none;
+  background-color: #FFF;
+  border: 2px solid #e9edf0;
+  border-radius: 5px;
+  display: block;
+  width: auto;
+  height: auto;
+  margin-top: 120px;
+  margin-bottom: 120px;
+  z-index: 9999999999999999;
```
Comment (peter-hank) [inline client/views/helpers/qr_code/qr_code.scss]: @singhsanket143 it's not a valid value https://stackoverflow.com/questions/491052/minimum-and-maximum-value-of-z-index/25461690#25461690

Comment (singhsanket143) [reply client/views/helpers/qr_code/qr_code.scss]: @peter-hank I have updated the z-index corresponding to chrome browser

File: client/views/helpers/nav/nav.js (hunk 7 above)
Comment (singhsanket143) [reply client/views/helpers/nav/nav.js]: @peter-hank Updated the spaces

File: .vscode/typings/meteor.d.ts (hunk 6 above)
Comment (singhsanket143) [reply .vscode/typings/meteor.d.ts]: @peter-hank I have Removed the following file

**Comment by singhsanket143** [comment]
> @singhsanket143 `public/.DS_Store` can you remove this file from the repository?

@peter-hank I have removed this file

**Comment by peter-hank** [comment]
@singhsanket143 ok, looks better now

**Pull request merged**